Assets/PatchKit Patcher/Scripts/AppUpdater/AppUpdaterContentStrategy.cs

```diff
@@ -64,7 +67,38 @@ public void Update(CancellationToken cancellationToken)
             installContent.Prepare(_status);
 
             uninstall.Execute(cancellationToken);
-            downloadContentPackage.Execute(cancellationToken);
+
+            var downloadStopwatch = new Stopwatch();
+            var optionalParams = new PatcherStatistics.OptionalParams 
+            {
+                VersionId = latestVersionId,
+                Size = resource.Size,
+            };
+
+            System.Func<PatcherStatistics.OptionalParams> timedParams = () => new PatcherStatistics.OptionalParams {
+                VersionId = optionalParams.VersionId,
+                Size = optionalParams.Size,
+                Time = downloadStopwatch.Elapsed.Seconds,
+            };
+
+            try
+            {
+                PatcherStatistics.DispatchSendEvent("content_download_started", optionalParams);
+                downloadStopwatch.Start();
+                downloadContentPackage.Execute(cancellationToken);
+                PatcherStatistics.DispatchSendEvent("content_download_succeeded", timedParams());
+            }
+            catch (System.OperationCanceledException)
+            {
+                PatcherStatistics.DispatchSendEvent("content_download_canceled", timedParams());
+                throw;
+            }
+            catch (System.Exception)
+            {
+                PatcherStatistics.DispatchSendEvent("content_download_failed", timedParams());
```

Comment:
What happens when an exception is thrown by DispatchSendEvent on line 89? Maybe it would be a good idea to wrap it differently?